### PR TITLE
Merge dev into master

### PR DESCRIPTION
**Pull request by Talasta**

### CODE REVIEW

File: day00/ex02/ex02.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+|                         |                    |
+| -----------------------:| ------------------ |
+|   Turn-in directory :   |  ex02              |
+|   Files to turn in :    |  variance.py       |
+|   Forbidden functions : |  *.sum(), np.mean(), np.var() |
+|   Remarks :             |  n/a               |
+
+## Objectives:
+
+You must implement the following formula as a function:
```
Comment (fxbabin) [inline day00/ex02/ex02.md]: maybe explain what is variance, why it is used (gaussian curve, std) ? or ask for a question ?

File: day00/ex07/ex07.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+|                         |                    |
+| -----------------------:| ------------------ |
+|   Turn-in directory :   |  ex07              |
+|   Files to turn in :    |  mse.py            |
+|   Forbidden functions : |  None              |
+|   Remarks :             |  n/a               |
+
+## Objectives:
+
+You must implement the following formula as a function:
```
Comment (fxbabin) [inline day00/ex07/ex07.md]: here also, maybe explain the purpose of mean square error ? (remark for future enhancement purpose)

File: day01/ex01/ex01.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
+# Exercise 01 - Cost function Concept
+
+|                         |                     |
+| -----------------------:| ------------------  |
+|   Turn-in directory :   |  ex01               |
+|   Files to turn in :    |  cost_function.py   |
+|   Authorized modules :  |  numpy,             |
```
Comment (fxbabin) [inline day01/ex01/ex01.md]: remove the comma ?

File: day01/ex02/ex02.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+## Remarks :
+
+You can generate other examples by choosing arbitrary X array and declare Y as linear expression of the X columns. Notice also that you can have the components of theta becoming "[nan]". In that case it means you probably used a too big learning rate.
+
+
+## Questions :
+
+Be sure to understand the underlying concepts and be able to answer those questions during your evaluation:
+* Can you explain the different steps in the fit method (hint: you have to talk about J, it gradient and the theta)?
```
Comment (fxbabin) [inline day01/ex02/ex02.md]: it gradient ???

File: day04/ex04/ex04.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Exercise 04 - Understanding better Decision Tree for Classification
+
+|                         |                    |
+| -----------------------:| ------------------ |
+|   Turn-in directory :   |  ex04              |
+|   Files to turn in :    |  answers.txt       |
+|   Forbidden modules :   |  NA                |
+|   Forbidden functions : |  NA                |
+|   Remarks :             |  Read the doc      |
+|   Helpful links         |  Statquest: https://www.youtube.com/watch?v=7VeUPuFGJHk&list=PLblh5JKOoLUICTaGLRoHQDuF_7q2GfuJF&index=34 |
```
Comment (fxbabin) [inline day04/ex04/ex04.md]: definitely use the shorter link syntax

File: day01/ex02/ex02.md (hunk 4 above)
Comment (matboivin) [reply day01/ex02/ex02.md]: Changed to 'the'

File: day01/ex01/ex01.md (hunk 3 above)
Comment (matboivin) [reply day01/ex01/ex01.md]: Done

File: day00/day00.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+The idea is to help the students to 'tame' the mathematical jargon. They will have to simply implement the functions following the formulas without knowing what it is about. Then magically this already 'tamed' material will be used in context and understood without having to fight with its abstract notation.
+
+## Notions of the day
+
+Sum, mean, variance, standard deviation, vectors and matrices operations.
+
+## Useful Ressources  
+  
+Machine Learning MOOC - Stanford:  
+https://www.coursera.org/learn/machine-learning/home/week/1
```
[the remark this answers is not in the document]
Comment (matboivin) [reply day00/day00.md]: Ok good idea :smile:

**Comment by matboivin** [comment]
I removed the hidden files and corrected some typos. I can fix the links by using a shorter link here with the notation like @fxbabin said. Would it be okay with the PDF builder? Or will it be even simpler for the next PDF builder tool?

**Comment by fxbabin** [comment]
the current pdf builder can take markdown links without problem (and the next one also) but not href ones. They must be formatted as followed : \[short-url\](long-url)

**Comment by matboivin** [comment]
@fxbabin  Url have been shortened as you explained! :smile:

File: day01/ex07/ex07.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+|                         |  alpha.py          |
+|   Authorized modules :  |  NumPy, Matplotlib |
+|   Forbidden modules :   |  sklearn           |
+|   Forbidden functions : |  LinearRegression  |
+|   Helpful links :       | Did I already give you a link to coursera Machine Learning MOOC? |
+|   Remarks :             | You know what to look at, I think... |
+
+## Objectives:
+
+* Learn other metrics such as RMSE and R2score, and understand there limits.
```
Comment (fxbabin) [inline day01/ex07/ex07.md]: there -> their

Comment (matboivin) [reply day01/ex07/ex07.md]: Fixed!

**Comment by matboivin** [comment]
:tada:

**Pull request merged**